changelogs/fragments/23-hashi-vault-lookup-refresh.yaml

```diff
@@ -0,0 +1,13 @@
+bugfixes:
+  - hashi_vault - when a non-token authentication method like ldap or userpass failed, but a valid token was loaded anyway (via env or token file), the token was used to attempt authentication, hiding the failure of the requested auth method.
+  - hashi_vault - if used via `with_hashi_vault` and a list of (say) 4 secrets to retrieve, only the first one would be retrieved and returned 4 times.
+  - hashi_vault - error messages are not user friendly and don't contain the secret name ( https://github.com/ansible-collections/community.general/issues/54 )
+minor_changes:
+  - hashi_vault - `secret` can now be an unnamed argument if it's specified first in the term string (see examples).
+  - hashi_vault - previously all options had to be supplied via key=value pairs in the term string; now a mix of string and parameters can be specified (see examples).
+  - hashi_vault - new option `return_format` added to control how secrets are returned, including options for multiple secrets and returning raw values with metadata.
+  - hashi_vault - `token` is now an explicit option (and the default) in the choices for `auth_method`
```

Comment:
What was the previous default value? Changing default values is always problematic, as it breaks backwards compatibility.

Reply:
I've updated the message to hopefully make this clearer, but I've gone to great lengths to preserve backwards compatibility for all of the changes in this PR.

In this case, the previous default was `None`, and the behavior from that was to use token auth (`token` was not in the `choices` before, you had to omit it).

So what I've changed is to add `token` as an explicit option in the choices, while also making it the default value, so this keeps previous _behavior_ and old playbooks will still work, but it now allows one to also use `auth_method: token` which previously would not have worked.

Let me know if the new changelog message is clear; if not I will not try to reword it.

Reply:
Sounds good to me!